### ISSUE TITLE
Font Linking for right-to-left scripts

### ISSUE THREAD

**Issue by cup** [mintty/mintty#807]
Control arm
-----------

1. install

       usr/share/fonts/noto/NotoSansSamaritan-Regular.ttf

   from

       noto-samaritan-fonts

1. Link font

       set 'HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\FontLink\SystemLink'
       reg add "$1" /t REG_MULTI_SZ /v Consolas /d NotoSansSamaritan-Regular.ttf

1. test file

       U+0800 SAMARITAN LETTER ALAF
       [ࠀ]

1. set font for `notepad.exe` to Consolas and test

1. Set font for `cmd.exe` to Consolas, `chcp 65001` and test

1. result success

Experimental arm
----------------

1. set font for `mintty.exe` to Consolas, `chcp.com 65001` and test

1. result failure

References
----------

- http://docs.microsoft.com/globalization/input/font-technology
- http://fileformat.info/info/unicode/char/0800/index.htm

**Comment by mintty** [comment]
> 4. set font for notepad.exe to Consolas and test
> 5. Set font for cmd.exe to Consolas, chcp 65001 and test

This does not display the glyph for me.

To confirm configuration:
reg query 'HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\FontLink\SystemLink' /v Consolas

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\FontLink\SystemLink
    Consolas    REG_MULTI_SZ    NotoSansSamaritan-Regular.ttf

There must be something else in your configuration that enables the glyph substitution.

**Comment by cup** [comment]
@mintty after registry you must relog - you dont have to restart but relog is required

**Comment by cup** [comment]
@mintty here is another test using left-to-right - note that for some reason
this new test has some different results:

- cmd.exe: success
- notepad: failure
- mintty: failure

Control arm
-----------

install:

~~~
usr/share/fonts/noto/NotoSansMyanmar-Regular.ttf
~~~

from:

~~~
noto-myanmar-fonts
~~~

Link font:

~~~sh
set 'HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\FontLink\SystemLink'
reg add "$1" /t REG_MULTI_SZ /v Consolas /d NotoSansMyanmar-Regular.ttf
~~~

test.txt:

~~~
U+1000 MYANMAR LETTER KA
[က]
~~~

Set font for `cmd.exe` to Consolas and test:

~~~
chcp 65001
type test.txt
~~~

Experimental arm
----------------

set font for `mintty.exe` to Consolas and test:

~~~sh
chcp.com 65001
cat test.txt
~~~

References
----------

http://fileformat.info/info/unicode/char/1000/index.htm

**Comment by mintty** [comment]
For Myanmar, mintty shows me glyphs, without extra configuration. For whatever reason you don't get them, it's likely unrelated.

**Comment by cup** [comment]
Mintty does not display Myanmar with Windows 7 - pristine virtual machine

**Comment by mintty** [comment]
You can test whether mintty shows Samaritan glyphs in your environment by forcing has_rtl = false in function win_text (wintext.c). Character order may be affected (force break_run = true in term.c to prevent that, with the side-effect of mintty becoming slow).

**Comment by cup** [comment]
@mintty unless you want to drop support for Windows 7

the title should probably be changed back to what it was

**Comment by mintty** [comment]
?
The problem was identified to occur only with right-to-left scripts and is caused by mintty special handling of this case. This has nothing to do with Windows 7.
Could you test what I suggested, by the way?

**Comment by mintty** [comment]
Released 3.0.2 with secondary font support. Configuration example:
```
Font9=Noto Sans Samaritan
FontChoice=Samaritan:9
```

**Comment by mintty** [comment]
For right-to-left scripts there is now a generic implicit font fallback mechanism: #1148
